pkg/controllers/user/rbac/handler_base.go

```diff
@@ -237,42 +235,24 @@ func (m *manager) ensureNamespacedRoles(rt *v3.RoleTemplate) error {
 }
 
 // isClusterOwner checks if a role template is cluster-owner, has cluster ownership rules, or inherits from a role template that grants cluster ownership.
-func (m *manager) isClusterOwner(rtName string) (bool, error) {
-	rt, err := m.rtLister.Get("", rtName)
-	if err != nil {
-		return false, err
-	}
-
+func (m *manager) isClusterOwner(rt *v3.RoleTemplate) (bool, error) {
 	// role template is the builtin cluster-owner
 	if rt.Builtin && rt.Context == "cluster" && rt.Name == "cluster-owner" {
 		return true, nil
 	}
 
-	// role template has rules for cluster ownership
 	for _, rule := range rt.Rules {
-		if slice.ContainsString(rule.Resources, "clusters") {
-			if slice.ContainsString(rule.Verbs, "own") {
-				return true, nil
-			}
+		// cluster + own rule that indicates cluster owner permissions
+		if slice.ContainsString(rule.Resources, "clusters") && slice.ContainsString(rule.Verbs, "own") {
+			return true, nil
 		}
-	}
-
-	isOwner := false
-	if len(rt.RoleTemplateNames) > 0 {
```

Comment:
why do we need to remove this?

Reply:
I was thinking that Cluster RoleTemplates can't have inherited role templates due to the option not being available in the UI, but they can if you add them through the api, so I will add this logic back in. Good catch